### ISSUE TITLE
Tests fail with mariadb-10.4.11: Access denied for user 'root'@'localhost' at lib/Test/mysqld.pm line 143

### ISSUE THREAD

**Issue by ppisar** [kazuho/p5-test-mysqld#32]
After upgrading mariadb from  10.3.21 to 10.4.11, all Test-mysqld tests fail like this:

~~~~
$ perl -Ilib t/01-raii.t 
DBI connect('dbname=mysql;mysql_socket=/tmp/cXEW5xhQn2/tmp/mysql.sock;user=root','',...) failed: Access denied for user 'root'@'localhost' at lib/Test/mysqld.pm line 143.
Access denied for user 'root'@'localhost' at lib/Test/mysqld.pm line 143.
~~~~

It seems to be triggered by a change in the default authentication settings <https://mariadb.com/kb/en/authentication-plugin-unix-socket/> brought with 10.4.3. Probably with this commit <https://github.com/MariaDB/server/commit/7f6d88944c>.

As a result mariadb uses unix socket for a local connection and enforces that the user specified with --user option is equaled to the user who runs the SQL client.

**Comment by ppisar** [comment]
I should notice that a current mariadb C client library has a broken UNIX socket authentication and a fix [https://jira.mariadb.org/browse/CONC-441?focusedCommentId=146491&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-146491] will be released in mariadb 10.4.13.

**Comment by Songmu** [comment]
Thanks!
